services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/ScanOperation.java

```diff
@@ -48,21 +59,42 @@ private ScanOperation(ScanEnhancedRequest request) {
     public ScanRequest generateRequest(TableSchema<T> tableSchema,
                                        OperationContext operationContext,
                                        DynamoDbEnhancedClientExtension extension) {
+        Map<String, AttributeValue> expressionValues = null;
+        Map<String, String> expressionNames = null;
+
+        if (this.request.filterExpression() != null) {
+            expressionValues = this.request.filterExpression().expressionValues();
+            expressionNames = this.request.filterExpression().expressionNames();
+        }
+
+        String projectionExpression = null;
+        if (!this.request.attributesToProject().isEmpty()) {
+            List<String> placeholders = new LinkedList<>();
```

Comment:
Can we use an ArrayList here as we're only adding and iterating?

Reply:
Sincerely, I barely use `LinkedList` (mostly because I live in a Groovy world where every new list `[]` is an `ArrayList`) but for the operations, you have also mentioned it looks like the best option to choose. As far as I can remember `ArrayList` is designed for access by index and `LinkedList` for adding and interacting.

Reply:
My best understanding is that LinkedList is beneficial if you need to insert records into the middle or start of a list, or if you wish to join two lists (and any other operations that require splicing and displacing the index of other records), whereas for this use case ArrayList is simpler, faster and uses less memory. It's even more efficient if you initialize the ArrayList with an accurate predicted capacity, but most people don't bother with this optimization.

Reply:
right, haven't thought about copying, just inserting.